lib/std/hash_map.zig

```diff
@@ -1378,16 +1357,11 @@ pub fn HashMapUnmanaged(
             return result;
         }
         pub fn getOrPutAssumeCapacityAdapted(self: *Self, key: anytype, ctx: anytype) GetOrPutResult {
-            comptime verifyContext(@TypeOf(ctx), @TypeOf(key), K, Hash, false);
 
             // If you get a compile error on this line, it means that your generic hash
             // function is invalid for these parameters.
             const hash = ctx.hash(key);
```

Comment:
This will make the compile error from wrong hash function return type much clearer:

```suggestion
            const hash: Hash = ctx.hash(key);
```